cancermuts/core.py

```diff
@@ -124,8 +130,10 @@ def __iter__(self):
         return iter(self.positions)
 
     def __repr__(self):
-        return "<Sequence of %s from %s, %d positions>" % (self.gene_id, self.source.name, len(self.positions))
-
+        form = "canonical" if self.is_canonical else "non_canonical"
+        iso_str = self.isoform if self.isoform else "no_isoform"
+        return f"<Sequence {self.gene_id} ({form}, isoform={iso_str}, uniprot_ac={self.uniprot_ac}) from {self.source.name}, {len(self.positions)} positions>"
```

Comment:
since this is the textual representation of an object, I think it's just fair to print the state of its attributes, e.g.:

```
<Sequence {self.gene_id} (uniprot_ac=..., isoform=..., is_canonical=..., from {self.source.name}, {len(self.positions)} positions>
```